packages/block-library/src/embed/index.js

```diff
@@ -183,10 +191,32 @@ export function getEmbedEdit( title, icon ) {
 				}
 
 				if ( aspectRatioClassName ) {
-					const className = classnames( this.props.attributes.className, 'wp-has-aspect-ratio', aspectRatioClassName );
-					this.props.setAttributes( { className } );
+					return {
+						[ aspectRatioClassName ]: true,
+						'wp-has-aspect-ratio': true,
+					};
 				}
 			}
+
+			return {};
+		}
+
+		/**
+		 * Maybe sets the appropriate CSS class names to enforce an aspect ratio when the embed
```

Comment:
Why does this "maybe" set things? I don't know this component very well, but this is a very confusing name/comment to me.

Reply:
Prefacing it with "Maybe" is confusing. It makes me think it might do something, but I can't tell the reason *why* it might do it 😄

I think the idea is it sets the `classNames` if `allowResponsive` is `true`.

In that case I think this is best to just be named `setAspectRatioClassNameIfResponsive`.

Alternatively we could rename it `setAspectRatioClassName` but I understand why that would be confusing; if `allowResponsive` is `false`-y we don't set anything, but the method name implies we are always changing a value.

I just find the "maybe" word confusing and not adding clarity. `setAspectRatioClassNameIfResponsive` is self-documenting. I think the method documentation could be cleaned up too, because it says stuff about `iframe` height/width, but in reality the check is around the `allowResponsive` attribute. We always set the `className` attribute if `allowResponsive` is truthy, it's *what* we set it to that varies on the iFrame props.

What about:

```js
		/**
		 * Set the CSS class names to enforce an aspect ratio for an embed
		 * that is allowed to resize. The aspect ratio is calculated from
		 * the iFrame inside the HTML provided.
		 *
		 * @param {string} html The preview HTML that possibly contains an iframe with width and height set.
		 */
```

Sorry, I know this documentation is tangential to your change but it's quite confusing to me and I think worth tidying up whilst we're changing things here.

Reply:
But setAttributes is always called if allowResponsive is true.

Reply:
(Sorry for terseness; I was on mobile when I left that comment!)

As talked about on Slack, I think `applyResponsiveClassnames` or `setResponsiveClassnames` would work here and just be a bit clearer/less confusing as a name 😄

Would that be okay?

Reply:
Sure! The maybe stuff had leaked through from me reading too much oEmbed code and I got slightly corrupted :rofl:

I'll get it changed around